### PR TITLE
hugo/feature/Change animation blinkOnCharge

### PR DESCRIPTION
**Pull request by HPezz**
- :children_crossing: (MotionKit): Improve smart reinforcer performance
- :recycle: (MotionKit): Add timeout to rotate() function
- :recycle: (libs): Update Motionkit calls
- :children_crossing: (BlinkOnCharge): Improve animation
- :children_crossing: (BehaviorKit): Trigger back BlinkOnCharge after BleConnection

### CODE REVIEW

File: libs/LedKit/include/animations/BlinkOnCharge.h [hunk 1]
```diff
@@ -45,6 +45,13 @@ class BlinkOnCharge : public interface::LEDAnimation
 	uint8_t _step		  = 0;
 	uint8_t _stage		  = 0;
 
+	void stage0();
+	void stage1();
+	void stage2();
```
Comment (ladislas) [inline libs/LedKit/include/animations/BlinkOnCharge.h:50]: can you name them something meaningful as you only have 3 and they do specific things

**Review by ladislas — CHANGES_REQUESTED** [review]
LGTM besides stage naming 👍

File: libs/BehaviorKit/source/BehaviorKit.cpp [hunk 2]
```diff
@@ -84,6 +84,9 @@ void BehaviorKit::bleConnection(bool with_video)
 	_ledkit.start(&led::animation::ble_connection);
 	if (with_video) {
 		_videokit.playVideoOnce("/fs/home/vid/system/robot-system-ble_connection-wink-no_eyebrows.avi");
+	} else {
+		rtos::ThisThread::sleep_for(5s);
+		blinkOnCharge();
```
Comment (YannLocatelli) [inline libs/BehaviorKit/source/BehaviorKit.cpp:89]: This function (bleConnection) does not have notion of charge, only of video.

This workaround is fine but bring some confusion in BehaviorKit implementation.

**Review by YannLocatelli — CHANGES_REQUESTED** [review]
Adding a charge notion in bleConnection is not good, and I just noticed that tests mention a condition on state connection or not: charging

State machine is composite, in the ideology to differentiate connection and robot states. So there are kind of not related.

Moreover Behaviors should not be aware of the state.

--

I don't have solution yet, I need some time to think about it.

File: libs/BehaviorKit/tests/BehaviorKit_test.cpp [hunk 3]
```diff
@@ -109,7 +109,6 @@ TEST_F(BehaviorKitTest, bleConnectionWhileCharging)
 TEST_F(BehaviorKitTest, bleConnectionWhileNotCharging)
 {
 	EXPECT_CALL(mock_videokit, playVideoOnce);
-	EXPECT_CALL(mock_ledkit, start(isSameAnimation(&led::animation::ble_connection))).Times(1);
```
Comment (YannLocatelli) [inline libs/BehaviorKit/tests/BehaviorKit_test.cpp:112]: Keep this one

**Review by YannLocatelli — CHANGES_REQUESTED** [review]
A typo + missing case

**Review by YannLocatelli — APPROVED** [review]

**Review by ladislas — APPROVED** [review]

**Pull request merged**